src/features/runtime-checks.js

```diff
@@ -573,4 +576,211 @@ export default class RuntimeChecks extends ContentFeature {
+        globalThis.Date = new Proxy(globalThis.Date, {
+            construct (target, args) {
+                const constructed = Reflect.construct(target, args)
+                if (getTaintFromScope(this, arguments, config.stackCheck)) {
+                    // Falible in that the page could brute force the offset to match. We should fix this.
+                    if (constructed.getTimezoneOffset() === offset) {
+                        return constructed.getUTCDate()
+                    }
+                }
+                return constructed
+            }
+        })
+    }
+
+    overloadDateGetTimezoneOffset (config) {
+        const offset = (new Date()).getTimezoneOffset()
+        defineProperty(globalThis.Date.prototype, 'getTimezoneOffset', {
+            configurable: true,
+            enumerable: true,
+            writable: true,
+            value () {
+                if (getTaintFromScope(this, arguments, config.stackCheck)) {
+                    return 0
+                }
+                return offset
+            }
+        })
+    }
+
+    overloadHighEntropyValues (config) {
+        if (!('NavigatorUAData' in globalThis)) {
+            return
+        }
+
+        const originalGetHighEntropyValues = globalThis.NavigatorUAData.prototype.getHighEntropyValues
+        defineProperty(globalThis.NavigatorUAData.prototype, 'getHighEntropyValues', {
+            configurable: true,
+            enumerable: true,
+            writable: true,
+            value (hints) {
+                let hintsOut = hints
+                if (getTaintFromScope(this, arguments, config.stackCheck)) {
+                    // If tainted override with default values (using empty array)
+                    hintsOut = []
+                }
+                return Reflect.apply(originalGetHighEntropyValues, this, [hintsOut])
+            }
+        })
+    }
+
+    overloadStorageWithMemory (config, key) {
+        class MemoryStorage {
+            #data = {}
+
+            setItem (id, val) {
+                this.#data[id] = String(val)
+            }
+
+            getItem (id) {
+                return Object.prototype.hasOwnProperty.call(this.#data, id) ? this.#data[id] : undefined
+            }
+
+            removeItem (id) {
+                return delete this.#data[id]
+            }
+
+            clear () {
+                this.#data = {}
+            }
+
+            key (n) {
+                const keys = Object.keys(this.#data)
+                return keys[n]
+            }
+
+            get length () {
+                return Object.keys(this.#data).length
+            }
+        }
+
+        const storage = new MemoryStorage()
+        this.overrideStorage(config, key, storage)
+    }
+
+    overloadStorageWithSession (key, config) {
+        const storage = globalThis.sessionStorage
+        this.overrideStorage(config, key, storage)
+    }
+
+    overrideStorage (config, key, storage) {
```

Comment:
Does this handle the following situations? 

```js
localStorage.foo = "bar"

console.log(localStorage.getItem('foo'))
// 'bar'
```

Reply:
Nope but it does now with the power of proxies and a test. Thanks for the catch here!